### DISCUSSION TITLE
Stash vector out of bounds from spurious loop?

### DISCUSSION THREAD

**Discussion by hateborne** [diasurgical/DevilutionX#5473] [General]
_(Hitting vector out of bounds issue with underflow itemId, which is what started all of this.)_

Why is the [InvDrawSlotBack](https://github.com/diasurgical/devilutionX/blob/680ab5ec40fcc2078ed6ad4a48bf081e68b1c9a1/Source/inv.cpp#L1020) call handled in a separate loop ([stash.cpp:355](https://github.com/diasurgical/devilutionX/blob/069ba7f1d4884a7abbe686db8827e1d97268ff1f/Source/qol/stash.cpp#L355)) instead of the loop right below it ([stash.cpp:363](https://github.com/diasurgical/devilutionX/blob/069ba7f1d4884a7abbe686db8827e1d97268ff1f/Source/qol/stash.cpp#L363))? The validation seems to be handled in the second loop and the first loop seems (from a relative outsider) problematic. 

Example of lines 355 - 381:
```cpp
	for (auto slot : StashGridRange) {
		StashStruct::StashCell itemId = Stash.GetItemIdAtPosition(slot);
		if (itemId == StashStruct::EmptyCell) {
			continue; // No item in the given slot
		}

		Item &item = Stash.stashList[itemId];

		if (Stash.IsItemAtPosition(slot)) {
			InvDrawSlotBack(out, GetStashSlotCoord(slot) + offset, InventorySlotSizeInPixels, item._iMagical);
		}

		if (item.position != slot) {
			continue; // Not the first slot of the item
		}

		int frame = item._iCurs + CURSOR_FIRSTITEM;

		const Point position = GetStashSlotCoord(item.position) + offset;
		const ClxSprite sprite = GetInvItemSprite(frame);

		if (pcursstashitem == itemId) {
			uint8_t color = GetOutlineColor(item, true);
			ClxDrawOutline(out, color, position, sprite);
		}

		DrawItem(item, out, position, sprite);
	}
```

**Comment by StephenCWills** [comment]
FYI, it would have been more appropriate to create an issue and follow the bug report template rather than creating a discussion. I'm creating a new issue from this discussion and will include a proper response there.

**Reply by hateborne** [reply]
Fair enough, I was trying to avoid pestering unnecessary and a discussion seemed (at the time) like a slightly less noisy way to go about it.

Thank you!
